packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -513,5 +513,6 @@ export const useField = <
     onMouseUp: handleInputMouseUp,
     error: inputError,
     ref: handleRef,
+    inputProps: { ...otherForwardedProps.inputProps, readOnly, disabled },
```

Comment:
Not sure to understand why we need to forward `disabled` through `inputProps`

`<TextField disabled />` is prefered over `<TextField inputProps={{ disabled }} />` so that the `FormControl` is also aware of it.

Am I missing something ?

___

For the `readOnly` I agree that the change is needed if we want the input to be aware of this prop

Reply:
The behaviour of disabled field and disabled field + input is different and I think a bit more correct.
Here is a video comparing the two. Let me know what you think 🤔 

https://user-images.githubusercontent.com/4941090/207817947-40854523-77d5-4a1b-89e6-05f966086f97.mov

Reply:
> The behaviour of disabled field and disabled field + input is different and I think a bit more correct.

When you are talking about "field", are you talking about `DateField` or `TextField` ?
I do agree that we need to forward the props to the `TextField`, I just don't understand why it should be through its `inputProps` prop.

Reply:
`field` -> `DateField`.
I'll try looking at the issue more closely and identify, why `disabled` attribute is not applied to `input`, once `TextField` is disabled. 😉

Reply:
Tested it and it looks like that this initial try & fail solution is no longer relevant after fixing the correct prop propagation chain. 💪 
Reverted. 😉

Reply:
Ahh, after latest test failures I can see that removing it introduces a regression of not applying `readonly` prop to the input element. 🤔 🙈

Reply:
The issue is that `TextField` does not support `readOnly` prop.
It is "officially" supported [via `InputProps`](https://mui.com/material-ui/react-text-field/#form-props). 🤔 
This requires some interesting solution, or getting back to my initial solution 🙈 😆